### PR TITLE
fix: 16300-rehaul-kubernetes

### PR DESCRIPTION
**Pull request by jnikhila**
The PR has fixes for [Installation Guide - Kubernetes](https://docs.appsmith.com/getting-started/setup/installation-guides/kubernetes/). Tracked under ticket [#16300](https://github.com/appsmithorg/appsmith/issues/16300).

- Rehaul Changes:
- Linter Issues
- Exception added to the style guide

### CODE REVIEW

**Comment by jnikhila** [comment]
@danciaclara - The markdown table do not have a fixed width and driven by content that is displayed. Because of which looks weird with varied widths on same page. I think, we can add some style changes to the same. Your thoughts?

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 1]
```diff
@@ -2,79 +2,95 @@
...
-2\. Install the chart with the <mark>`appsmith`</mark> release.
+2. Install the chart with the <mark>`appsmith`</mark> release.
 
-```
+```bash
 helm install appsmith/appsmith --generate-name
 ```
 
-The command deploys the Appsmith application on the Kubernetes cluster in the default configuration. The [Parameters](https://github.com/appsmithorg/appsmith/tree/release/deploy/helm#paramters) section lists the configurable parameters during installation.
+The command deploys Appsmith on the Kubernetes cluster with the default configuration. Read more on [configurations](#configuration) that you could do during installation.
```
Comment (danciaclara) [inline website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:77]: Do we really need to cross-reference the Configuration section, which is right below the Install on the same page? You can maybe rewrite this as

> The command deploys the Appsmith application on the Kubernetes cluster with the default configuration. You can configure the chart using the parameters shown in the following section.

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 2]
```diff
@@ -2,79 +2,95 @@
...
-```
-helm list
-NAME                       NAMESPACE       REVISION        UPDATED                                 STATUS          CHART           APP VERSION
-appsmith-1631069261        default         1               2021-09-09 11:24:40.152766 +0700 +07    deployed        appsmith-1.3.0  1.16.0
-
-helm uninstall appsmith-1631069261
-```
-
-The command uninstalls the release and removes all Kubernetes resources associated with the chart.
+## Configuration
```
Comment (danciaclara) [inline website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:80]: Let's stay consistent with the headings. In the above section, it starts with 'Install', so here it should be 'Configure the Helm chart'

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 3]
```diff
@@ -163,82 +179,81 @@ The command uninstalls the release and removes all Kubernetes resources associat
...
-#### Auto-update chart's image
+#### Auto update image
 
 | Name                   | Description                                   | Value         |
 | ---------------------- | --------------------------------------------- | ------------- |
 | `autoupdate.enabled`   | Enable auto update Helm chart's image         | `true`        |
 | `autoupdate.scheduler` | Schedule time to run cron job to update image | `"0 * * * *"` |
 
-Specify each parameter using `--set key=value[,key=value]` argument to helm install. For example:
+### Configure parameters
```
Comment (danciaclara) [inline website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:189]: there shouldn't be a separate section again. Why can't we put this right before the table of parameters under the main Configure heading followed by the parameters list?

we should spread this content properly so there is a flow and there is no break in the information leading to confusion

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 4]
```diff
@@ -163,82 +179,81 @@ The command uninstalls the release and removes all Kubernetes resources associat
...
-```
-helm install \
---set applicationConfig.APPSMITH_ENCRYPTION_SALT=123 \
-  stable-appsmith/appsmith --generate-name
+```bash
+helm install -f values.yaml stable-appsmith/appsmith --generate-name
 ```
 
-## Expose Appsmith
+## Publish
```
Comment (danciaclara) [inline website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:243]: this heading should either be 'Expose Appsmith server' or 'Publish Appsmith server'

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 5]
```diff
@@ -249,17 +264,19 @@ helm install \
 
 ## Updating Appsmith
 
-### Auto Update
+Appsmith installation can be updated manually or by enabling auto update.
+
+### Auto update
 
-*   In the default Appsmith helm installation the auto-update is disabled (Recommended). You have the option to enable auto-update for your Appsmith helm deployment by either:
+The default Appsmith helm installation disables the auto update (_Recommended_). You have an option to enable auto update by either setting it in values file or using a command.
```
Comment (danciaclara) [inline website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:271]: Should the 'recommended' word be on the heading itself?

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 6]
```diff
@@ -269,39 +286,51 @@ helm install \
...
-* To manually update the Appsmith container image to the latest release, run the command: `kubectl rollout restart statefulset appsmith`
+To manually update the Appsmith container image to the latest release, run the command: 
 
-## Nginx ingress controller compression (Gzip)
+```bash
+kubectl rollout restart statefulset appsmith
+```
 
-In the default installation, the Nginx ingress controller will not have the data compression configured. Please follow the following steps to enable it.
+## Uninstall
```
Comment (danciaclara) [inline website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:297]: Let this be 'Uninstall Helm chart' and move it to the bottom, just before Troublshooting

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 7]
```diff
@@ -269,39 +286,51 @@ helm install \
...
+```bash
+helm list
+NAME                       NAMESPACE       REVISION        UPDATED                                 STATUS          CHART           APP VERSION
+#highlight-next-line
+appsmith-1631069261        default         1               2021-09-09 11:24:40.152766 +0700 +07    deployed        appsmith-1.3.0  1.16.0
 
+helm uninstall appsmith-1631069261
 ```
+
+## Enable Ingress data compression
```
Comment (danciaclara) [inline website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:309]: This should be 'Compress Nginx Ingress controller'

**Review by danciaclara — CHANGES_REQUESTED** [review]

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 8]
```diff
@@ -2,79 +2,95 @@
...
+import TabItem from '@theme/TabItem';
 
 # Kubernetes
 
-## Installing Appsmith using Helm Charts
+[Kubernetes[K8s]](https://kubernetes.io/docs/concepts/overview/) is an open source platform for managing, scaling, and automating the deployment of [containerized applications](https://en.wikipedia.org/wiki/Containerization). It's one of the most popular container orchestration tools. The guide helps you to install [Appsmith](https://github.com/appsmithorg/appsmith) on a [Kubernetes](https://github.com/appsmithorg/appsmith/blob/release/deploy/helm/kubernetes.io) cluster using [Helm](https://helm.sh) package manager.
 
-This chart bootstraps an [Appsmith](https://github.com/appsmithorg/appsmith) deployment on a [Kubernetes](https://github.com/appsmithorg/appsmith/blob/release/deploy/helm/kubernetes.io) cluster using [Helm](https://helm.sh) package manager.
+## Prerequisites
+Follow the guides below to set up Helm and `kubectl`.
```
[the remark this answers is not in the document]
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:13]: Omit the `Follow the guides below`.

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 1 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:77]: Yes, we can remove this.

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 9]
```diff
@@ -2,79 +2,95 @@
...
+<TabItem label="Microk8s" value="Microk8s">
+
+  1. Set up Kubernetes by following [working with kubectl](https://microk8s.io/docs/working-with-kubectl).
+  2. Ensure that you have a default storage class running. If not, [enable storage class](https://microk8s.io/docs/command-reference#heading--microk8s-enable).
+  3. The NGINX Ingress Controller should be enabled by default. If not, [add on: Ingress](https://microk8s.io/docs/addon-ingress). Ensure that you enable the correct version for your cluster.
+    
+</TabItem>
+</Tabs>
+
+## Install
```
[the remark this answers is not in the document]
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:53]: This is Installing Appsmith on Helm Chart. The Helm chart is installed as part of the Prerequisites section.

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 10]
```diff
@@ -2,79 +2,95 @@
...
+
+1. Click to download the [`values.yaml`](https://bit.ly/3ETEgPT) file, and place it into the Appsmith installation folder.
```
[the remark this answers is not in the document]
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:55]: Okay.

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 2 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:80]: I can reword it to configure

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 3 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:189]: I think the parameter table gives a view of all the parameters which the users can configure, and then the description below shows how to configure them. I felt adding a separate section will help readability.

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 4 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:243]: I think Publish Appsmith. Why we want to use server?

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 5 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:271]: No, we are not recommending auto-update. The auto-update is disabled by default, which is recommended. Thus, I kept it there only. I didn't add `recommended` to the Manual update and kept it the way it was in the original doc.

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 6 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:297]: This section uninstalls the Appsmith installation and not the Helm chart. I added it here as it completes the docs flow.

**Comment by danciaclara** [comment]
> @danciaclara - The markdown table do not have a fixed width and driven by content that is displayed. Because of which looks weird with varied widths on same page. I think, we can add some style changes to the same. Your thoughts?

@jnikhila Hey didn't understand the context here. Can you share some screenshots?

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 9 above)
[the remark this answers is not in the document]
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:53]: As discussed - I will add `Install Appsmith`

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 3 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:189]: As discussed, I'll add a few lines of explanation followed by the table and keep the examples below the table.

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 4 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:243]: As discussed - I will add `Publish Appsmith`

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 5 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:271]: As discussed, no changes needed here.

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 6 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:297]: As discussed - I will add `Uninstall Appsmith` and move the section just before Troubleshooting.

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 7 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:309]: As discussed - I will add - `Enable data compression for Ingress controller`

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 10 above)
[the remark this answers is not in the document]
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:55]: > > @danciaclara - The markdown table do not have a fixed width and driven by content that is displayed. Because of which looks weird with varied widths on same page. I think, we can add some style changes to the same. Your thoughts?
> 
> @jnikhila Hey didn't understand the context here. Can you share some screenshots?

As discussed - I can look into having a custom CSS for markdown tables. However, that's not a blocking change for this PR.
